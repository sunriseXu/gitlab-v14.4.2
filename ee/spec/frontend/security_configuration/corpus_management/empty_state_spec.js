import { shallowMount } from '@vue/test-utils';

import EmptyState from 'ee/security_configuration/corpus_management/components/empty_state.vue';

const TEST_CORPUS_HELP_PATH = '/docs/corpus-management';
const TEST_EMPTY_STATE_SVG_PATH = '/illustrations/no_commits.svg';

describe('EE - CorpusManagement - EmptyState', () => {
  let wrapper;
  const testButton = '<button>Perform action</button>';

  const createComponent = (options = {}) => {
    wrapper = shallowMount(EmptyState, {
      provide: {
        corpusHelpPath: TEST_CORPUS_HELP_PATH,
        emptyStateSvgPath: TEST_EMPTY_STATE_SVG_PATH,
      },
      slots: {
        actions: testButton,
      },
      ...options,
    });
  };

  it('should render correct content', () => {
    createComponent();

    expect(wrapper.element).toMatchSnapshot();
  });
});
