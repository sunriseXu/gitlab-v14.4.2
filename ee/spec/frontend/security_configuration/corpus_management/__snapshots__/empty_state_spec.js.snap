// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EE - CorpusManagement - EmptyState should render correct content 1`] = `
<gl-empty-state-stub
  invertindarkmode="true"
  svgpath="/illustrations/no_commits.svg"
  title="Manage your fuzz testing corpus files"
>
  <gl-sprintf-stub
    message="A corpus is used by fuzz testing to improve coverage. Corpus files can be manually created or auto-generated. %{linkStart}Learn more%{linkEnd}"
  />
  <button>
    Perform action
  </button>
</gl-empty-state-stub>
`;
